AdSecGH/Helpers/BusinessGlue.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+
+using AdSecCore.Functions;
+
+using Grasshopper.Kernel;
+
+using OasysGH;
+using OasysGH.Components;
+
+namespace Oasys.GH.Helpers {
+
+  public abstract class ComponentAdapter<T> : GH_OasysComponent, IDefaultValues where T : IFunction {
+
+    public readonly T BusinessComponent = Activator.CreateInstance<T>();
+
+    public ComponentAdapter() : base("", "", "", "", "") {
+      Name = BusinessComponent.Metadata.Name;
+      NickName = BusinessComponent.Metadata.NickName;
+      Description = BusinessComponent.Metadata.Description;
+      Category = BusinessComponent.Organisation.Category;
+      SubCategory = BusinessComponent.Organisation.SubCategory;
+    }
+
+    public override OasysPluginInfo PluginInfo { get; } = AdSecGH.PluginInfo.Instance;
+    public void SetDefaultValues() { BusinessComponent.SetDefaultValues(this); }
+
+    protected override void RegisterInputParams(GH_InputParamManager pManager) {
+      BusinessComponent.PopulateInputParams(this);
+    }
+
+    protected override void RegisterOutputParams(GH_OutputParamManager pManager) {
+      BusinessComponent.PopulateOutputParams(this);
+    }
+
+    protected override void SolveInstance(IGH_DataAccess DA) {
+      BusinessComponent.UpdateInputValues(this, DA);
+      BusinessComponent.Compute();
+      BusinessComponent.SetOutputValues(this, DA);
+    }
+  }
+
+  public interface IDefaultValues {
+    void SetDefaultValues();
+  }
+
+  public abstract class DropdownAdapter<T> : GH_OasysDropDownComponent, IDefaultValues where T : IFunction {
```

Comment:
@psarras I do not have any strong reservations if something is working, but people generally follow one project file (.cs) per class.

Reply:
agree